tutorials/src/bin/basic-tutorial-9.rs

```diff
@@ -0,0 +1,150 @@
+extern crate failure;
+use failure::Error;
+
+extern crate glib;
+extern crate gstreamer as gst;
+extern crate gstreamer_pbutils as pbutils;
+use pbutils::prelude::*;
+use pbutils::Discoverer;
+use pbutils::DiscovererContainerInfo;
+use pbutils::DiscovererInfo;
+use pbutils::DiscovererResult;
+use pbutils::DiscovererStreamInfo;
+
+use std::env;
+
+fn print_stream_info(info: &DiscovererStreamInfo, depth: usize) {
+    print!("{:indent_level$}", " ", indent_level = 2 * depth);
+    print!("{}:", info.get_stream_type_nick());
+
+    if let Some(caps) = info.get_caps() {
+        let desc = if caps.is_fixed() {
+            pbutils::pb_utils_get_codec_description(&caps)
+                .unwrap_or_else(|| String::from("unknown codec"))
+        } else {
+            caps.to_string()
+        };
+        println!(" {}", desc);
+    } else {
+        println!();
+    }
+
+    if let Some(tags) = info.get_tags() {
+        println!("{:indent_level$}Tags:", " ", indent_level = 2 * depth);
+        tags.iter_tag_list().for_each(|(tag, values)| {
```

Comment:
Instead of the `for_each`, often it is nicer to do

```rust
for (tag, values) in tags.iter_tag_list() {
    ....
}
```

You could also use `tags.iter()` now, which gives you a merged value for each tag.